keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModelBuilder.java

```diff
@@ -14,17 +15,19 @@
  * @see RegressionModel
  */
 public class RegressionModelBuilder<OUTPUT extends Tensor> {
+
     private static final double DEFAULT_MU = 0.0;
     private static final double DEFAULT_SCALE_PARAMETER = 1.0;
 
-    private RegressionRegularization regularization = RegressionRegularization.NONE;
-    private double[] priorOnWeightsScaleParameters;
-    private double[] priorOnWeightsMeans;
-    private Double priorOnInterceptScaleParameter;
-    private Double priorOnInterceptMean;
     private final DoubleTensor inputTrainingData;
     private final OUTPUT outputTrainingData;
```

Comment:
If you no longer do the fitting inside the builder, you don't need these fields or the `withTrainingData` method

Reply:
I prefer defining the complete data set / fitting inside the builder, or else we have to `build` and then continue passing more parameters which feels incorrect

Reply:
Agreed - we are on the same page here.